twill-core/src/main/java/org/apache/twill/kafka/client/KafkaConsumer.java

```diff
@@ -33,14 +33,16 @@
 
     /**
      * Invoked when new messages is available.
+     * @param startOffset Offset of the first {@link FetchedMessage} in the iterator of new messages.
```

Comment:
Is it? Or shall it return the offset of the first message being read? Other, e.g. if `-1L` was specified, and the latest message was read, but not being consumed by the `onReceived` method. If it returns `-1L`, then that particular message won't get fetched again if there is a newer message available. Is that the intention?